codegenerator/cli/src/fuel/abi.rs

```diff
@@ -112,214 +91,165 @@ impl Abi {
-                            let type_ident_name = mk_type_id_name(&comp.type_id);
-                            let type_ident = match &comp.type_arguments {
-                                //When there are no type arguments it is a named type or a generic param
-                                None => generic_param_name_map.get(&comp.type_id).cloned().map_or(
-                                    //If the type_id is not a defined generic type it is
-                                    //a named type
-                                    RescriptTypeIdent::TypeApplication {
-                                        name: type_ident_name,
-                                        type_params: vec![],
-                                    },
-                                    //if the type_id is in the generic_param_name_map
-                                    //it is a generic param
-                                    |generic_name| RescriptTypeIdent::GenericParam(generic_name),
-                                ),
-                                //When there are type arguments it is a generic type
-                                Some(typ_args) => {
-                                    let type_params = typ_args
-                                        .iter()
-                                        .map(|ta| {
-                                            generic_param_name_map.get(&ta.type_id).cloned().map_or(
-                                                //If the type_id is not a defined generic type it is
-                                                //a named type
-                                                RescriptTypeIdent::TypeApplication {
-                                                    name: mk_type_id_name(&ta.type_id),
-                                                    type_params: vec![],
-                                                },
-                                                //if the type_id is in the generic_param_name_map
-                                                //it is a generic param
-                                                |generic_name| {
-                                                    RescriptTypeIdent::GenericParam(generic_name)
-                                                },
-                                            )
-                                        })
-                                        .collect();
-                                    RescriptTypeIdent::TypeApplication {
-                                        name: type_ident_name,
-                                        type_params,
-                                    }
-                                }
-                            };
-                            Ok((name, type_ident))
-                        })
-                        .collect::<Result<Vec<_>>>()
-                };
+            let type_id = TypeId::Metadata(MetadataTypeId(type_decl.type_id));
```

Comment:
Are all types in the program metadata? I seem to remember there were concrete types as well.